datamodel/test/core_tests.cpp

```diff
@@ -0,0 +1,63 @@
+#include <gtest/gtest.h>
+
+#include <rcbe-engine/datamodel/core/CoreObject.hpp>
+#include <rcbe-engine/datamodel/core/Dimensions.hpp>
+#include <rcbe-engine/fundamentals/types.hpp>
+
+inline constexpr rcbe::core::EngineIntergral IWIDTH = 1024;
+inline constexpr rcbe::core::EngineIntergral IHEIGHT = 768;
+
+namespace {
+struct some_empty_struct {};
+}
+
+TEST(DimensionsTests, IntegerDimensions) {
+    rcbe::core::Dimensions dim{IWIDTH, IHEIGHT};
+
+    ASSERT_EQ(dim.width, IWIDTH);
+    ASSERT_EQ(dim.height, IHEIGHT);
+}
+
+TEST(DimensionsTests, FloatDimensions) {
+    using FloatDimensions = rcbe::core::dimensions_impl<float>;
+    static constexpr float FWIDTH = static_cast<float>(IWIDTH);
+    static constexpr float FHEIGHT = static_cast<float>(IHEIGHT);
+    FloatDimensions fdim{FWIDTH, FHEIGHT};
+
+    ASSERT_EQ(fdim.width, FWIDTH);
+    ASSERT_EQ(fdim.height, FHEIGHT);
+}
+
+TEST(DimensionsTests, DoubleDimensions) {
+    static constexpr float DWIDTH = static_cast<rcbe::core::EngineScalar>(IWIDTH);
+    static constexpr float DHEIGHT = static_cast<rcbe::core::EngineScalar>(IHEIGHT);
+    rcbe::core::ScalarDimensions sdim{DWIDTH, DHEIGHT};
+
+    ASSERT_EQ(sdim.width, DWIDTH);
+    ASSERT_EQ(sdim.height, DHEIGHT);
+}
+
+TEST(CoreObjectTests, Components) {
+    using DimT = rcbe::core::Dimensions;
+
+    rcbe::core::CoreObject co{some_empty_struct{}};
+
+    DimT dim{IWIDTH, IHEIGHT};
```

Comment:
This thing is created two times, consider introducing a fixture for storing this. Also, consider underlying type conversion, for above test cases.

Reply:
Fixed